src/application/clients/crux-wallet-client.ts

```diff
@@ -455,14 +471,21 @@ export class CruxWalletClient {
         }
         return selfClaim;
     }
-    private setupCruxMessenger = async (selfIdClaim: ICruxIdClaim | undefined) => {
+    private setupCruxMessenger = async (selfIdClaim: ICruxIdClaim | undefined, options: ICruxWalletClientOptions) => {
         if (!selfIdClaim) {
             throw Error("Self ID Claim is required to setup messenger");
         }
+        if (options && options.disableCruxMessenger) {
```

Comment:
You're doing this check in both setupCruxMessenger and asyncInit - pick one

Reply:
done